wrapper/src/test/java/integration/container/tests/TopologyQueryTests.java

```diff
@@ -0,0 +1,275 @@
+import integration.container.condition.EnableOnNumOfInstances;
+import integration.util.AuroraTestUtility;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.MethodOrderer.MethodName;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import software.amazon.jdbc.dialect.AuroraMysqlDialect;
+import software.amazon.jdbc.dialect.AuroraPgDialect;
+import software.amazon.jdbc.dialect.Dialect;
+import software.amazon.jdbc.dialect.RdsMultiAzDbClusterMysqlDialect;
+import software.amazon.jdbc.dialect.RdsMultiAzDbClusterPgDialect;
+
+@TestMethodOrder(MethodName.class)
+@DisableOnTestFeature({
+    TestEnvironmentFeatures.PERFORMANCE,
+    TestEnvironmentFeatures.RUN_HIBERNATE_TESTS_ONLY,
+    TestEnvironmentFeatures.RUN_AUTOSCALING_TESTS_ONLY})
+public class TopologyQueryTests {
+  private static final Logger LOGGER = Logger.getLogger(TopologyQueryTests.class.getName());
+  private String query = null;
+
+  @TestTemplate
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  @ExtendWith(TestDriverProvider.class)
+  public void testAuroraTypes(TestDriver testDriver) throws SQLException {
+    LOGGER.info(testDriver.toString());
+    List<String> expectedTypes;
+
+    final Properties props = ConnectionStringHelper.getDefaultPropertiesWithNoPlugins();
+    DriverHelper.setConnectTimeout(testDriver, props, 10, TimeUnit.SECONDS);
+    DriverHelper.setSocketTimeout(testDriver, props, 10, TimeUnit.SECONDS);
+
+    String url =
+        ConnectionStringHelper.getWrapperUrl(
+            testDriver,
+            TestEnvironment.getCurrent()
+                .getInfo()
+                .getDatabaseInfo()
+                .getInstances()
+                .get(0)
+                .getHost(),
+            TestEnvironment.getCurrent()
+                .getInfo()
+                .getDatabaseInfo()
+                .getInstances()
+                .get(0)
+                .getPort(),
+            TestEnvironment.getCurrent().getInfo().getDatabaseInfo().getDefaultDbName());
+
+    if (TestEnvironment.getCurrent().getCurrentDriver() == TestDriver.MARIADB
+        && TestEnvironment.getCurrent().getInfo().getRequest().getDatabaseEngine() == DatabaseEngine.MYSQL) {
+      props.setProperty("permitMysqlScheme", "1");
+    }
+
+    LOGGER.finest("Connecting to " + url);
+
+    if (TestEnvironment.getCurrent().getCurrentDriver() == TestDriver.PG) {
+      AuroraPgDialect dialect = new AuroraPgDialect();
```

Comment:
I think you could move this if-else to the beginning of the test after `List<String> expectedTypes;`, before establishing a connection, since you don't need any connections here and you are just setting up for the test.